### PR TITLE
Add tooling environment and consolidate manifests.

### PR DESCRIPTION
**Pull request by jmcarp**
This changes the structure of the secrets a bit to collapse the tooling / staging / production manifests, so we'll need to update secrets if this gets merged.

Also, not sure that we need separate tooling / staging / production branches here--can we push everything that varies between environments to the secrets?

@LinuxBozo

### CODE REVIEW

File: ci/pipeline.yml [hunk 1]
```diff
@@ -12,13 +16,69 @@ groups:
...
+        username: {{slack-username}}
+        icon_url: {{slack-icon-url}}
+
+- name: deploy-monitoring-staging
+  plan:
+  - aggregate:
     - get: monitoring-staging-config
       trigger: true
     - get: common-stage
+    - get: common-tooling
```
Comment (LinuxBozo) [inline ci/pipeline.yml]: Why are you getting common-tooling here for deploying staging?

File: ci/pipeline.yml [hunk 2]
```diff
@@ -73,10 +128,11 @@ jobs:
   - aggregate:
     - get: monitoring-staging-deployment
       passed: [deploy-monitoring-staging]
-    - get: pipeline-tasks
     - get: monitoring-production-config
       trigger: true
     - get: common-prod
+    - get: common-tooling
```
Comment (LinuxBozo) [inline ci/pipeline.yml]: Again, why getting common-tooling if this is for deploying prod?

File: monitoring.yml [hunk 3]
```diff
@@ -150,7 +153,7 @@ properties:
       - database_type: "influxdb"
         database_name: (( meta.influxdb.database ))
         url: (( "http://" jobs.influxdb.networks.monitoring.static_ips.[0] ":8086" ))
-        name: "influxdb_production"
+        name: (( meta.grafana.name ))
```
Comment (LinuxBozo) [inline monitoring.yml]: Shouldn't this be `meta.influxdb.name`?

Comment (jmcarp) [inline monitoring.yml]: That would make sense--I was just looking down a few lines and saw `meta.grafana.influxdb_user` and `meta.grafana.influxdb_password`, which is why it turned out like this. Think we can get rid of `meta.grafana.influxdb_*` and keep the influxdb properties in `meta.influxdb`?

File: ci/pipeline.yml (hunk 1 above)
Comment (jmcarp) [inline ci/pipeline.yml]: Sorry, because I don't know how to copy and paste right.

File: monitoring.yml (hunk 3 above)
Comment (LinuxBozo) [inline monitoring.yml]: Yeah, this seems ripe for refactoring. There is also `riemann.influxdb_user`, etc as well..

**Comment by jmcarp** [comment]
Updated!

**Comment by LinuxBozo** [comment]
@jmcarp So, for branches, we've been using this as a stopgap to make sure that production always = `master`, but we may want to experiment in staging, so, `staging` branch.. this means all pulls go against `staging` instead of `master`. In this case, where there are three environments, this model obviously breaks down, unless tooling is considered like production, and you just use `master` as well. The other option is to do like you pointed out, which is use secrets, and make sure that you are using dependencies within the deployment pipeline (using `passed:` to control)

**Comment by jmcarp** [comment]
It looks like the only real difference between `staging` and `master` is that we've only been pushing compliance updates (encryption, snort) to `master`. I would vote for reconciling `staging` and `master` to avoid confusion, and since that seems to be what we're doing on other deploy repos.

**Comment by jmcarp** [comment]
All right, I verified that this branch includes staging and updated the pipeline so that all three environments use the same git source, and so that production doesn't deploy until staging succeeds.

**Pull request merged**